src/Base/Core/Quality/IQuality.hs

```diff
@@ -10,8 +10,21 @@ Portability : POSIX
 This module exports the (interval) Quality datatype and a
 function to get the base quality of an interval size.
 -}
+
+{-# LANGUAGE PatternSynonyms #-}
+
 module Base.Core.Quality.IQuality
-  ( Quality(..)
+  ( Quality
+  , pattern Major
+  , pattern Minor
+  , pattern Perfect
+  , pattern Diminished
+  , pattern Augmented
```

Comment:
These exports expose _only_ pattern forms of the constructors for this type; this way, we and other users can pattern-match in a natural way, but _can't_ use the underlying representation. Note that I removed `(..)` from the export of `Quality`; this keeps all the actual constructors hidden.